### ISSUE TITLE
azure_rm_adgroup_info returns error if device is not member of group

### ISSUE THREAD

**Issue by mhallin2** [ansible/ansible_collections_azure#1575]
##### SUMMARY
Checking if a Azure DeviceID is member of a group fails if device is not a member of the group.
but no issues if the device is a member

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure.azcollection.azure_rm_adgroup_info:

##### ANSIBLE VERSION

core 2.16.5

```

##### COLLECTION VERSION
<!--- Paste verbatim output from "ansible-galaxy collection list <namespace>.<collection>"  between the quotes
for example: ansible-galaxy collection list community.general
-->
Collection         Version
------------------ -------
azure.azcollection 2.3.0  

```

##### CONFIGURATION

Not possible due to running Ansible automation platform in Azure Manange Application so do not have console

```

##### OS / ENVIRONMENT
N/A


##### STEPS TO REPRODUCE
<!--- Describe exactly how to reproduce the problem, using a minimal test-case -->
Trying to check if Device is member of a Azure AD Security Group

<!--- Paste example playbooks or commands between quotes below -->
```yaml
    - name: Get Azure AD group information
      azure.azcollection.azure_rm_adgroup_info:
        tenant: "{{ AzureTenant }}"
        attribute_name: "displayName"
        attribute_value: "{{ AAD_Group_Name }}"
        check_membership: "{{ DeviceID.json.value[0].id }}"
      register: Add_DeviceID_To_AAD_Group_Info
```

##### EXPECTED RESULTS

The expected result is that the module should not report an error

##### ACTUAL RESULTS

 "msg": "failed to get ad group info 
APIError
Code: 404
message: 
error: MainError(additional_data={}, 
code='Request_ResourceNotFound', 

```paste below
  "msg": "failed to get ad group info \n        APIError\n        Code: 404\n        message: None\n        error: MainError(additional_data={}, code='Request_ResourceNotFound', details=None, inner_error=InnerError(additional_data={'date': DateTime(2024, 5, 26, 7, 12, 29, tzinfo=Timezone('UTC'))}, client_request_id='<REMOVED>', date=None, odata_type=None, request_id='REMOVED'), message=\"Resource '<GoupID >' does not exist or one of its queried reference-property objects are not present.\", target=None)\n        ",
  "invocation": {
    "module_args": {
      "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
      "attribute_name": "displayName",
      "attribute_value": "GRoupNAME",
      "check_membership": "<DeviceID>",
      "auth_source": "auto",
      "cloud_environment": "AzureCloud",
      "api_profile": "latest",
      "disable_instance_discovery": false,
      "return_owners": false,
      "return_group_members": false,
      "return_member_groups": false,
      "all": false,
      "profile": null,
      "subscription_id": null,
      "client_id": null,
      "secret": null,
      "ad_user": null,
      "password": null,
      "cert_validation_mode": null,
      "adfs_authority_url": null,
      "log_mode": null,
      "log_path": null,
      "x509_certificate_path": null,
      "thumbprint": null,
      "object_id": null,
      "odata_filter": null
    }
  },
  "_ansible_no_log": false,
  "changed": false
}
```

**Comment by Fred-sun** [comment]
@mhallin2 Yes, we are now in such nursing logic that if the relationship between the two is detected, it will prompt such an error, and if it is true, the result returned is as follows. Also, we will make such changes that if the relationship does not hold, ""is_member_of": false" will be returned without an error. Thank you!
```e.g.
            {
                "description": "for test",
                "display_name": "xxxxxxxxxxxxxxxx",
                "is_member_of": true,
                "mail": null,
                "mail_enabled": false,
                "mail_nickname": "fredtest001-Group-Root",
                "object_id": "xxxxxxxxxxxxxxxxxxxxxxxx",
                "security_enabled": true
            }
            {
                "description": "for test",
                "display_name": "fredtest001-Group-Root",
                "is_member_of": false,
                "mail": null,
                "mail_enabled": false,
                "mail_nickname": "fredtest001-Group-Root",
                "object_id": "xxxxxxxxxxxxxxxxxxxxxxxx,
                "security_enabled": true
            }

**Comment by Fred-sun** [comment]
fixes by #1579
